main/src/ca/uwaterloo/flix/language/phase/CodeGen.scala

```diff
@@ -100,13 +100,14 @@ object CodeGen extends Phase[ExecutableAst.Root, ExecutableAst.Root]{
     // 3. Create Enum Type info
     val allEnums : List[(Type, (String, Type))] = root.definitions.values.flatMap(x => CodegenHelper.findEnumCases(x.exp)).toList
 
-    val enumTypeInfo: Map[(Type, String), QualName] = allEnums.map{ case (tpe, (name, subType)) =>
+    val enumTypeInfo: Map[(Type, String), (QualName, Boolean)] = allEnums.map{ case (tpe, (name, subType)) =>
```

Comment:
`ExecutableAst.Case` doesn't help that much here except for finding if it is a singleton or not.
This map is used to find the right class that implements the enum and we need type of the field to do it yet case doesn't provide this info.
Best thing I can do is to change it to `Map[(Type, String), (QualName, Boolean)]`. Do you prefer this?